Workflows/GIT/Branches.md

```diff
@@ -0,0 +1,34 @@
+Our Git process is an upgraded version of [Github Flow](https://guides.github.com/introduction/flow/).
+
+![](https://ftp.infinum.co/stjepan_hadjic/git-flow-2.jpg)
+
+Instead of a single `master` branch, we use two branches to record the history of the project. The `master` branch stores code that's currently in production. The `staging` branch serves as an integration branch for features and fixes. It holds data that reflects what's currently on staging.
+
+**The master branch must always be deployable.**
+
+As we use [Productive](https://productive.io) to manage our tasks, we use **feature branches** for fixes and new features. Our branch naming convention is {type}/{task-number}{descriptive-task-name}. For example, a branch for adding authentication would be named `feature/123-authentication`.
+
+*Types*
+
+* _feature_ - new feature or an improvement to an existing feature.
+* _fix_ - non-critical bugfix, improvement, paying the technical debt. Goes through the code review process.
+* _hotfix_ - time sensitive critical bugfix, that should be deployed to production as soon as possible. It is not necessary for it to go through a code review, but it should be revisited at a later stage, and properly fixed or improved.
+
+Feature branches are **always** branched out of `master`. They are also merged first into `staging`, and then into `master` if they are ready for production.
```

Comment:
I'd say that they SHOULD always be branched out of master, but not always. Sometimes you need something from a feature branch and you don't want to wait for it to be merged into Master, maybe because of something that is delaying that branch, so it would make sense to open a PR with a temporary base set to the feature branch you based on, to provide a clean difffor the PR. 
Note that this flow requres extra attentiom when merging into master as you have to change the base branch if the PR and be cautious when rebasing the feature branch in case the base branche’s history changed.

I would not include all of this in the handbook, I’d just change the wording to SHOULD so that it is not so strict 😊